modules/channel-telegram/src/renderers/carousel.ts

```diff
@@ -0,0 +1,45 @@
+import * as sdk from 'botpress/sdk'
+import path from 'path'
+import { TelegramContext } from 'src/backend/typings'
+import { CallbackButton, Markup } from 'telegraf'
+import Extra from 'telegraf/extra'
+
+export class TelegramCarouselRenderer implements sdk.ChannelRenderer<TelegramContext> {
+  getChannel(): string {
+    return 'telegram'
+  }
+
+  getPriority(): number {
+    return 0
+  }
+
+  getId() {
+    return TelegramCarouselRenderer.name
+  }
+
+  async handles(context: TelegramContext): Promise<boolean> {
+    return context.payload.type === 'carousel'
+  }
+
+  async render(context: TelegramContext): Promise<void> {
+    const { messages } = context
+    const payload = context.payload as sdk.CarouselContent
+
+    if (payload.items?.length) {
+      const { title, image, subtitle } = payload.items[0]
+      const buttons = payload.items.map(x => x.actions || [])
+
+      if (image) {
+        messages.push({ action: 'upload_photo' })
```

Comment:
What's this ?

Reply:
Basically what I'm doing here is building the message as the exact format that telegram accepts but not actually sending it. That way, another renderer with a lower priority could add something/ modify this message before it is sent.

The choices renderer for example has a lower priority and adds a keyboard extra to any messages that requires it before being sent. Another example of this would be the feedback renderer in the slack channel which adds a feedback block to any message that requires it before being sent.